### PR TITLE
Fixing header suite failures

### PR DESCRIPTION
**Pull request by rakatyal**
case 1. Add a new type definition.
- [ ] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [ ] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [ ] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

**Comment by mhegazy** [comment]
Thank you for your contribution. 

Can you please fix the failing Travis CI tests and update the PR!

File: react-router/lib/createMemoryHistory.d.ts [hunk 1]
```diff
@@ -1,3 +1,8 @@
+// Type definitions for react-router v2.0.0
```
Comment (RyanCavanaugh) [inline react-router/lib/createMemoryHistory.d.ts]: I think this pattern is problematic. What happens if these headers don't agree within a folder? How do people know which to update? How is the types publisher supposed to know which version to use when generating the readme?

The header should only be required for the entry point file.

Comment (rakatyal) [inline react-router/lib/createMemoryHistory.d.ts]: Yup sounds good. I will update the definition-tester and then update the PR with the remaining errors.

**Comment by rakatyal** [comment]
@mhegazy: The CI server is running the old definition tester code which is expected to break against the 2.0 PRs.

**Comment by mhegazy** [comment]
sorry.. just marked it without looking.

**Comment by rakatyal** [comment]
Updated after modifying the definition-tester code to run for only the entry point files.

**Pull request closed without merging**